src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java

```diff
@@ -1,31 +1,85 @@
 package uk.ac.ebi.eva.submission.service;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import uk.ac.ebi.eva.submission.model.Submission;
 import uk.ac.ebi.eva.submission.model.SubmissionStatus;
 import uk.ac.ebi.eva.submission.repository.SubmissionRepository;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.UUID;
 
 @Service
 public class SubmissionService {
 
     private SubmissionRepository submissionRepository;
 
+    @Value("${eva.submission.dropbox}")
+    private String submissionDropbox;
+
     public SubmissionService(SubmissionRepository submissionRepository) {
         this.submissionRepository = submissionRepository;
     }
 
-    public SubmissionStatus createFtpDirectory() {
-        String submissionId = UUID.randomUUID().toString();
+    public Submission initiateSubmission() {
+        String submissionId = createSubmissionDirectory();
+
+        Submission submission = new Submission(submissionId);
+        submission.setStatus(SubmissionStatus.OPEN.toString());
+        submission.setInitiationTime(LocalDateTime.now());
+
+        return submissionRepository.save(submission);
+    }
+
+    public Submission markSubmissionUploaded(String submissionId) {
+        Submission submission = submissionRepository.findBySubmissionId(submissionId);
+        submission.setStatus(SubmissionStatus.UPLOADED.toString());
+        submission.setUploadedTime(LocalDateTime.now());
+
+        return submissionRepository.save(submission);
+    }
 
-        SubmissionStatus submissionStatus = new SubmissionStatus(submissionId);
-        return submissionRepository.save(submissionStatus);
+    public String getSubmissionStatus(String submissionId) {
+        Submission submission = submissionRepository.findBySubmissionId(submissionId);
+        return submission.getStatus();
     }
 
-    public SubmissionStatus markSubmissionComplete(String submissionId) {
-        SubmissionStatus submissionStatus = submissionRepository.findBySubmissionId(submissionId);
-        submissionStatus.setCompleted(true);
+    public Submission markSubmissionStatus(String submissionId, SubmissionStatus status) {
+        Submission submission = submissionRepository.findBySubmissionId(submissionId);
+        submission.setStatus(status.toString());
+        if (status == SubmissionStatus.COMPLETED) {
+            submission.setCompletionTime(LocalDateTime.now());
+        }
```

Comment:
Should this set the other timestamps for the other statuses (`OPEN` and `UPLOADED`) as well, or do we assume that the other endpoints will be used to set those?

Reply:
Yes, those are being set through different end-points